### DISCUSSION TITLE
Track internalization into intracellular when implementing TMDD

### DISCUSSION THREAD

**Discussion by rudgeman** [Open-Systems-Pharmacology/Forum#1463] [How-Tos] [answered]
Hi all,

I have implemented TMDD with a protein binding partner in interstitial space using the usual method (add synthesis/degradation and internalization of the complex). My question is, how can I set up a way to track the amount of complex that gets transported into the intracellular space via the complex internalization? If I right click and "insert molecule" using the reactions diagram in MoBi, I can connect to a reaction but get a warning that a reactant is not present.

My thought was to create a dummy molecule initialized to 0 to use to track this amount. In the end I would ask:

1. Is there a more simple way to track this value than a dummy molecule?
2. Is it possible to have this reaction move from the interstitial space into the intracellular (thus getting the intracellular concentration)
3. As a final side question, when Tissue or Whole Organ concentrations are calculated, do they include drug-receptor complexes, or only unbound drug? If the latter, I assume an observer would need to be defined to track the sum of all drug and complex values.

Any help or recommendations is very appreciated!

**Comment by Yuri05** [comment, marked as answer]
>Is there a more simple way to track this value than a dummy molecule?

The easiest way which comes into my mind would be:
a) Activate ["Create process rate" parameter](https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/model-building-components#reactions-and-molecules) for the reaction/transport of interest.
b) Create a new parameter with initial value 0 and define it as a state variable.
c) Set the right hand side (RHS) of this parameter either to a particular process rate parameter (if you want to track **just one single reaction or transport**) or to the sum of process rates for all processes of interest
d) Set this parameter to plottable

Check the example project: [Reaction_Transfer.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/10852799/Reaction_Transfer.zip)
Here, a reaction A2B is defined in 2 containers which (reversibly) transforms molecule A into molecule B.
In the organism, 2 parameters are defined: the first one tracks one particular reaction, the 2nd all reactions
<details><summary><b>Activate reaction process rate parameter</b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/221926432-2c76e35b-c439-4748-b30f-9a182576a209.png)

</details>
<details><summary><b>Tracking 1 reaction</b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/221925545-0cbbdf8d-01a8-4d8d-bf84-7ea52d49dc84.png)

</details>

<details><summary><b>Tracking N reactions</b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/221925942-47d9a0de-c9a2-493e-8a81-9ff0715806b6.png)
![grafik](https://user-images.githubusercontent.com/25061876/221926149-a3dac191-5816-4dfe-866f-a40d1d4cb01d.png)

</details>
<details><summary><b>Plot transferred amounts</b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/221927124-b3de359a-7c40-49a5-88fe-4c7e0e219ba3.png)

</details>

>Is it possible to have this reaction move from the interstitial space into the intracellular (thus getting the intracellular concentration)

This can be done by a passive transport. Reactions always take place within the **same container** per design.

>when Tissue or Whole Organ concentrations are calculated, do they include drug-receptor complexes, or only unbound drug? If the latter, I assume an observer would need to be defined to track the sum of all drug and complex values.

You assumption is correct: complexes are not included and you would need to define an extra observer for this.

**Reply by rudgeman** [reply]
Thank you very much for the illustrative reply! It made the process very clear.

A couple follow-ups seeing it done this way.

First, would there be a way to define this parameter relatively/generically so that it occurred in all containers the reaction did and was available for plotting? So the highlighted portion would instead read something like ..|A2B|ProcessRate

![image](https://user-images.githubusercontent.com/35583446/221936628-95870e7d-bfa0-47de-b2e2-5b64dc57b948.png)


Second, for passive transport, can this occur using the parameter implementation, or would the dummy molecule be required to implement that?

**Reply by Yuri05** [reply]
>First, would there be a way to define this parameter relatively/generically so that it occurred in all containers the reaction did and was available for plotting? So the highlighted portion would instead read something like ..|A2B|ProcessRate

Yes. You would need to define a parameter as a **local reaction parameter.**

<details><summary><b>Track transfer of the parent reaction </b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/221957479-432e5f28-55ee-4e82-bc05-43e10c528cbc.png)
</details>

>Second, for passive transport, can this occur using the parameter implementation, or would the dummy molecule be required to implement that?

You can do it in the same way as for reactions.
(Please note: if you do it for transports via the local transport parameter like above, you would need to show the Type column via the [column chooser](https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/features-of-tables#select-columns) first - it is hidden as per default here)

**Reply by rudgeman** [reply]
I see, important to do this under Reactions vs Spatial Structure. I ended up using the dummy molecule as it made a bit more sense (to me at least) on tracking the amounts.

I can also then add degradation in the IC by adding a reaction with a Container Criteria to limit it only to the IC if I am not mistaken.

Thank you very much for the help Yuri!
